Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+            let platforms = optionalRow("Verified Platforms", version.verifiedPlatforms?.map { $0.name }.joined(separator: ", "))
+            let swiftVersions = optionalRow("Verified Swift Versions", version.verifiedSwiftVersions?.map { $0.rawValue }.joined(separator: ", "))
+            let license = optionalRow("License", version.license?.type.description)
+            let cves = optionalRow("CVEs", version.cves?.map { $0.identifier }.joined(separator: ", "))
+
+            return """
+                \(version.version)
+                Package Name: \(version.packageName)
+                Modules: \(modules)\(platforms)\(swiftVersions)\(license)\(cves)
+            """
+        }
+
+        mutating func run() throws {
+            guard let packageUrl = packageUrl else {
+                throw CollectionsError.missingArgument("packageUrl")
+            }
+
+            let identity = PackageReference.computeIdentity(packageURL: packageUrl)
+            let reference = PackageReference(identity: identity, path: packageUrl)
+
+            let result = try await { self.collections.getPackageMetadata(reference, profile: profileOptions.usedProfile, callback: $0) }
+
+            if let versionString = version {
+                guard let version = TSCUtility.Version(string: versionString), let result = result.package.versions.first(where: { $0.version == version }), let printedResult = printVersion(result) else {
+                    throw CollectionsError.invalidVersionString(versionString)
+                }
+
+                print("Version: \(printedResult)")
+            } else {
+                let description = optionalRow("Description", result.package.description)
+                let versions = result.package.versions.map { "\($0.version)" }.joined(separator: ", ")
+                let watchers = optionalRow("Watchers", result.package.watchersCount?.description)
+                let readme = optionalRow("Readme", result.package.readmeURL?.absoluteString)
+                let authors = optionalRow("Authors", result.package.authors?.map { $0.username }.joined(separator: ", "))
+                let latestVersion = optionalRow("--------------------------------------------------------------\nLatest Version", printVersion(result.package.latestVersion))
+
+                print("""
+                    \(description)Available Versions: \(versions)\(watchers)\(readme)\(authors)\(latestVersion)
+                """)
+            }
+        }
+    }
+}
+
+private func optionalRow(_ title: String, _ contents: String?) -> String {
+    if let contents = contents {
+        return "\n\(title): \(contents)\n"
+    } else {
+        return ""
+    }
+}
+
+private extension JSONEncoder {
+    func print<T>(_ value: T) throws where T : Encodable {
+        if #available(macOS 10.15, *) {
+            self.outputFormatting = [.prettyPrinted, .sortedKeys, .withoutEscapingSlashes]
```

Comment:
I think prettyPrinted was supported prior to 10.15?

Reply:
I think so as well.  Also, we're accumulating a lot of cases where we check the macOS availability for this particular purpose (specifically for JSON formatting options).  I wonder if this is worth collecting into one place where we apply the standard formatting that we always seem to want (basically "as readable as possible").

Reply:
Yes, it is `.withoutEscapingSlashes` that isn't. I don't think it's worth the complexity of multiple availability checks since SwiftPM does not run on any macOS older than 10.15 anyway, but I can break it up into multiple statements if desired.

Reply:
If SwiftPM requires macOS 10.15 then we should just make that the minimum deployment target in the package manifest and have one #if check.

Reply:
We can do that, but so far we intentionally avoid it to allow clients of libSwiftPM to use 10.10. We set 10.15 in `bootstrap` (see https://github.com/apple/swift-package-manager/blob/main/Utilities/bootstrap#L28), so the actual CLI tools have a 10.15 minimum deployment target.

Reply:
> but so far we intentionally avoided it to allow clients of libSwiftPM to use 10.10

are there specific clients in mind?

Reply:
https://github.com/apple/swift-package-manager/pull/2710#issuecomment-618731081

Reply:
This will be addressed by adopting https://github.com/apple/swift-package-manager/pull/3071

Reply:
I think this ^^ still stand. we should change to use the makeWIthDefaults and get rid f the setter here?

Reply:
Yep, thanks for noticing.